storage/sftp/storage_sftp.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * knoxite
+ *     Copyright (c) 2019, Fabian Siegel <fabians1999@gmail.com>
+ *
+ *   For license see LICENSE
+ */
+
+package sftp
+
+import (
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/url"
+
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+
+	knoxite "github.com/knoxite/knoxite/lib"
+)
+
+type StorageSFTP struct {
+	url   url.URL
+	ssh   *ssh.Client
+	sftp  *sftp.Client
+	login bool
+	knoxite.StorageFilesystem
+}
+
+var (
+	ErrInvalidAuthentication = errors.New("Wrong Username or Password")
+)
+
+func init() {
+	knoxite.RegisterBackendFactory(&StorageSFTP{})
+}
+
+func (*StorageSFTP) NewBackend(u url.URL) (knoxite.Backend, error) {
+
+	_, port, err := net.SplitHostPort(u.Host)
+	if err != nil || len(port) == 0 {
+		port = "22"
+		u.Host = net.JoinHostPort(u.Host, port)
+	}
+	username := u.User.Username()
+	password, isSet := u.User.Password()
+
+	auth := []ssh.AuthMethod{}
+	if isSet {
+		auth = append(auth, ssh.Password(password))
+	}
+
+	config := &ssh.ClientConfig{
+		User:            username,
+		Auth:            auth,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
```

Comment:
Should think about how we handle that properly.

Reply:
Added better handling for this usecase, but we still need a solution for non-unix systems.